### PR TITLE
Fix downstream dependencies for `spl-program-error`

### PR DESCRIPTION
**Pull request by buffalojoec**
This PR fixes the requirement for crates that wish to use `spl-program-error` to have to import `thiserror`, `num-derive`, and `num-traits`.

Note: This will NOT require these crates to be present in the Cargo.toml file to use the **procedural macro attribute** `#[spl_program_error]`, however, if you decide to use any of the **derive macros** you'll need to import these crates.

### CODE REVIEW

File: libraries/program-error/derive/src/macro_impl.rs [hunk 1]
```diff
@@ -104,6 +104,7 @@ pub fn spl_program_error(input: ItemEnum) -> proc_macro2::TokenStream {
     let print_program_error = print_program_error(ident, variants);
     quote! {
         #[derive(Clone, Debug, Eq, thiserror::Error, num_derive::FromPrimitive, PartialEq)]
+        #[num_traits = "num_traits"]
```
Comment (joncinque) [inline libraries/program-error/derive/src/macro_impl.rs:107]: Ah very clever, this was the bit that I missed, nice job!

**Review by joncinque — APPROVED** [review]
perfect!

File: libraries/program-error/derive/src/macro_impl.rs (hunk 1 above)
Comment (buffalojoec) [reply libraries/program-error/derive/src/macro_impl.rs:107]: Turns out I just had to read their docs 😭

**Comment by buffalojoec** [comment]
Ran a rebase from `master`, looks like didn't really even need to

**Review by joncinque — APPROVED** [review]

**Pull request merged**